packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-comment-box.tsx

```diff
@@ -71,72 +69,115 @@ const PullRequestCommentBox = ({
+  }
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0]
+    if (file) {
+      handleUploadCallback(file)
+    }
+  }
+
+  const handleDragEnter = (e: React.DragEvent) => {
+    e.preventDefault()
+    e.stopPropagation()
+
+    if (e.currentTarget === dropZoneRef.current) {
+      setIsDragging(true)
+    }
+  }
+
+  const handleDragLeave = (e: React.DragEvent) => {
+    e.preventDefault()
+    e.stopPropagation()
+
+    if (e.currentTarget === dropZoneRef.current && !e.currentTarget.contains(e.relatedTarget as Node)) {
+      setIsDragging(false)
+    }
+  }
+
+  const handleDrop = (e: React.DragEvent) => {
+    e.preventDefault()
+    e.stopPropagation()
+    setIsDragging(false)
+    handleDropForUpload(e)
+  }
+
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const handleDropForUpload = async (event: any) => {
     handleFileDrop(event, handleUploadCallback)
   }
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const handlePasteForUpload = (event: { preventDefault: () => void; clipboardData: any }) => {
+
+  const handlePasteForUpload = (event: React.ClipboardEvent) => {
     handlePaste(event, handleUploadCallback)
   }
 
+  // TODO: add the remaining required logic for the toolbar
+  const toolbar: ToolbarItem[] = useMemo(() => {
+    const initial: ToolbarItem[] = []
+    return [
+      ...initial,
+      { icon: 'suggestion', action: ToolbarAction.SUGGESTION },
+      { icon: 'header', action: ToolbarAction.HEADER },
+      { icon: 'bold', action: ToolbarAction.BOLD },
+      { icon: 'italicize', action: ToolbarAction.ITALIC },
+      { icon: 'attachment', action: ToolbarAction.UPLOAD, onClick: handleFileSelect },
+      { icon: 'list', action: ToolbarAction.UNORDER_LIST },
+      { icon: 'checklist', action: ToolbarAction.CHECK_LIST },
+      { icon: 'code', action: ToolbarAction.CODE_BLOCK }
+    ]
+  }, [])
+
   return (
-    <div className="flex items-start space-x-4">
-      {(!isEditMode || !hideAvatar) && avatar}
+    <div className="flex items-start gap-x-3">
+      {!inReplyMode && avatar}
+
       <div
-        className={cn('min-w-0 flex-1 px-4 pb-5 pt-1.5', {
+        className={cn('pb-5 pt-1.5 px-4 flex-1 bg-background-2 border-border-1', {
           'border rounded-md': !inReplyMode || isEditMode,
-          'border-t ': inReplyMode
+          'border-t': inReplyMode
         })}
       >
         <Tabs variant="tabnav" defaultValue="write">
           <TabsList className="relative left-1/2 w-[calc(100%+32px)] -translate-x-1/2 px-4">
-            <TabsTrigger value="write">Write</TabsTrigger>
-            <TabsTrigger value="preview">Preview</TabsTrigger>
+            <TabsTrigger className="data-[state=active]:bg-background-2" value="write">
+              Write
+            </TabsTrigger>
+            <TabsTrigger className="data-[state=active]:bg-background-2" value="preview">
+              Preview
```

Comment:
![image](https://github.com/user-attachments/assets/a2a5023f-198d-4594-bf20-9f76da8af713)
in preview, the background changes for the content and it kinda looks weird
@andrewgolovanov